sycl/include/sycl/queue.hpp

```diff
@@ -149,6 +149,36 @@ class __SYCL_EXPORT SubmissionInfo {
       ext::oneapi::experimental::event_mode_enum::none;
 };
 
+using KernelParamDescGetterFuncPtr = detail::kernel_param_desc_t (*)(int);
+
+// This class is intended to store the kernel runtime information,
+// extracted from the compile time kernel structures.
+class __SYCL_EXPORT KernelRuntimeInfo {
```

Comment:
please define copy/move ctor and assignment operator. I guess they can be declared as deleted, right?